src/NHSD.BuyingCatalogue.Solutions.API/Controllers/AdditionalServiceController.cs

```diff
@@ -10,39 +15,44 @@ namespace NHSD.BuyingCatalogue.Solutions.API.Controllers
     [Produces(MediaTypeNames.Application.Json)]
     public sealed class AdditionalServiceController : ControllerBase
     {
+        private readonly IMediator _mediator;
+
+        public AdditionalServiceController(IMediator mediator)
+        {
+            _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator)); ;
+        }
+
         [HttpGet]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Cannot be viewed on swagger if static")]
-        public ActionResult<IEnumerable<AdditionalServiceResult>> Get([FromQuery] string[] solutionIds)
+        public async Task<ActionResult<IEnumerable<AdditionalServiceResult>>> GetAsync([FromQuery] IEnumerable<string> solutionIds)
         {
             if (solutionIds is null)
+            {
                 return NotFound();
+            }
 
-            var additionalServiceResults = new List<AdditionalServiceResult>();
-
-            foreach (var solutionId in solutionIds)
+            solutionIds = solutionIds.ToList();
+            if (!solutionIds.Any())
             {
-                additionalServiceResults.Add(new AdditionalServiceResult
-                {
-                    AdditionalServiceId = SetAdditionalServiceId(solutionId),
-                    Name = "Write on Time Additional Service 1",
-                    Summary = "Addition to Write on Time",
-                    Solution = new AdditionalServiceSolutionResult
-                    {
-                        SolutionId = "100000-001",
-                        Name = "Write on Time"
-                    }
-                });
+                return NotFound();
             }
 
-            return additionalServiceResults;
-        }
+            var additionalServices = (await _mediator.Send(new GetAdditionalServiceBySolutionIdsQuery(solutionIds))).ToList();
 
-        private static string SetAdditionalServiceId(string solutionId)
-        {
-            var additionalServiceId = string.Empty;
-            additionalServiceId += $"{solutionId}A001";
+            if (additionalServices.Count is 0)
+                return NotFound();
```

Comment:
I'm not sure that we should return NotFound here, an empty list is probably better. NotFound implies the solution ids weren't real, but they were... they just didn't have any additional services. Hmm. It's a tricky one. Might be worth checking with @sam-roberts-bjss

Reply:
Yeah Ruth & I talked about this & agreed to return an empty list. Also I think it should return all additional services if no solution ID is supplied. Agreed to raise this as tech debt, there isn't a requirement for this yet.